src/main/java/com/denizenscript/clientizen/objects/ItemTag.java

```diff
@@ -0,0 +1,146 @@
+package com.denizenscript.clientizen.objects;
+
+import com.denizenscript.clientizen.util.Utilities;
+import com.denizenscript.denizencore.objects.*;
+import com.denizenscript.denizencore.objects.properties.PropertyParser;
+import com.denizenscript.denizencore.tags.Attribute;
+import com.denizenscript.denizencore.tags.ObjectTagProcessor;
+import com.denizenscript.denizencore.tags.TagContext;
+import com.denizenscript.denizencore.utilities.debugging.Debug;
+import net.minecraft.item.ItemConvertible;
+import net.minecraft.item.ItemStack;
+import net.minecraft.registry.Registries;
+import net.minecraft.util.Identifier;
+
+public class ItemTag implements ObjectTag, Adjustable {
+
+    // <--[ObjectType]
+    // @name ItemTag
+    // @prefix i
+    // @base ElementTag
+    // @implements PropertyHolderObject
+    // @ExampleTagBase client.self_entity.item_in_hand
+    // @ExampleValues <client.self_entity.item_in_hand>,stick,iron_sword
+    // @ExampleForReturns
+    // - narrate "The item is %VALUE%"
+    // @format
+    // The identity format for items is the basic material type name. Other data is specified in properties.
+    // For example, 'i@stick'.
+    //
+    // @description
+    // An ItemTag represents a holdable item generically.
+    //
+    // ItemTags are temporary objects, to actually modify an item in an inventory you must add the item into that inventory.
+    //
+    // ItemTags do NOT remember where they came from. If you read an item from an inventory, changing it
+    // does not change the original item in the original inventory. You must set it back in.
+    //
+    // Find a list of valid materials at <@link url https://hub.spigotmc.org/javadocs/spigot/org/bukkit/Material.html>
+    // Note that some materials on that list are exclusively for use with blocks, and cannot be held as items.
+    //
+    // -->
```

Comment:
- Format uses a MaterialTag identifier
- You can't change an item on a client permanently like you can on the server
- Does not use Bukkit `Material` enum

Reply:
> Format uses a MaterialTag identifier

Not sure what are you referring to?
> You can't change an item on a client permanently like you can on the server

Yeah fair enough, will fix up the wording there a bit but assuming we have like, items in GUIs or whatever that explanation still stands.
> Does not use Bukkit `Material` enum

That's probably the best reference still? it's meant to perfectly match what the server-side `ItemTag` is doing - I guess we could maybe use <https://maven.fabricmc.net/docs/yarn-1.19.4+build.2/net/minecraft/item/Items.html>?

Reply:
You can probably ignore the first point but I wanted to put emphasis on `Identifier`s.

Reply:
Well, it'd still be `stick` either way, as the `minecraft:` namespace is stripped - improved the wording there a little bit though, lmk what do you think